src/applications/vaos/actions/newAppointment.js

```diff
@@ -111,6 +117,63 @@ export function openFacilityPage(page, uiSchema, schema) {
   };
 }
 
+export function openTypeOfFacilityPage(page, uiSchema, schema, router) {
+  const self = this;
+  return async (dispatch, getState) => {
+    const newAppointment = getState().newAppointment;
+    let systems = newAppointment.systems;
+
+    // fetch systems
+    if (!systems) {
+      dispatch({
+        type: FORM_PAGE_TYPE_OF_FACILITY_OPEN,
+        page,
+        uiSchema,
+        schema,
+      });
+
+      const identifiers = await getSystemIdentifiers();
+      const systemIds = identifiers
+        .filter(id => id.assigningAuthority.startsWith('dfn'))
+        .map(id => id.assigningCode);
+
+      systems = await getSystemDetails(systemIds);
+    }
+
+    // compare systems
+    const communityCareSites = await getSitesSupportingVAR();
+    const x = communityCareSites.find(site =>
+      systems.find(userSite => userSite.assigningCode === site._id),
+    );
+    if (x === undefined) {
+      dispatch(
+        updateFormData(page, uiSchema, {
+          facilityType: 'vamc',
+          vaFacility: '',
+          typeOfCareId: '323',
+        }),
+      );
+      // eslint-disable-next-line no-use-before-define
+      self.routeToNextAppointmentPage(router, 'vaFacility');
```

Comment:
This should be `typeOfFacility` instead of `vaFacility`. I forgot that we are supposed to pass the current pageKey we are on, not the one we want to go to.